modules/openthread/Kconfig

```
@@ -257,7 +257,8 @@ config OPENTHREAD_CUSTOM_PARAMETERS
 
 config OPENTHREAD_NUM_MESSAGE_BUFFERS
 	int "The number of message buffers in the buffer pool"
-	default 128
+	default 128 if !OPENTHREAD_ZEPHYR_BORDER_ROUTER
+	default 512 if OPENTHREAD_ZEPHYR_BORDER_ROUTER
```

A little simpler IMHO (first successful match gets preference)
```suggestion
	default 512 if OPENTHREAD_ZEPHYR_BORDER_ROUTER
	default 128
```